core/src/main/java/bisq/core/trade/protocol/tasks/seller/SellerSendsDepositTxAndDelayedPayoutTxMessage.java

```diff
@@ -17,85 +17,164 @@
 
 package bisq.core.trade.protocol.tasks.seller;
 
+import bisq.core.network.MessageState;
 import bisq.core.trade.Trade;
 import bisq.core.trade.messages.DepositTxAndDelayedPayoutTxMessage;
-import bisq.core.trade.protocol.tasks.TradeTask;
-
-import bisq.network.p2p.NodeAddress;
-import bisq.network.p2p.SendMailboxMessageListener;
+import bisq.core.trade.messages.TradeMessage;
+import bisq.core.trade.protocol.tasks.SendMailboxMessageTask;
 
+import bisq.common.Timer;
+import bisq.common.UserThread;
 import bisq.common.taskrunner.TaskRunner;
 
-import org.bitcoinj.core.Transaction;
+import javafx.beans.value.ChangeListener;
 
-import java.util.UUID;
+import java.util.concurrent.TimeUnit;
 
 import lombok.extern.slf4j.Slf4j;
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
+/**
+ * We send the buyer the deposit and delayed payout tx. We wait to receive a ACK message back and resend the message
+ * in case that does not happen in 4 seconds or if the message was stored in mailbox or failed. We keep repeating that
+ * with doubling the interval each time and until the MAX_RESEND_ATTEMPTS is reached. If never successful we fail and
+ * do not continue the protocol with publishing the deposit tx. That way we avoid that a deposit tx is published but the
+ * buyer does not has the delayed payout tx and would not be able to open arbitration.
+ */
 @Slf4j
-public class SellerSendsDepositTxAndDelayedPayoutTxMessage extends TradeTask {
-    @SuppressWarnings({"unused"})
-    public SellerSendsDepositTxAndDelayedPayoutTxMessage(TaskRunner taskHandler, Trade trade) {
+public class SellerSendsDepositTxAndDelayedPayoutTxMessage extends SendMailboxMessageTask {
```

Comment:
This resend type class has a lot of duplicate code from `BuyerSendCounterCurrencyTransferStartedMessage`, could probably use a common base class, or helper class.

Reply:
The SendMailboxMessageTask base class is already handling most. The exta code for resend has some differences so the same part is not worth IMO to extract to another superclass. If you compate both classes you see that there are small differences.

Reply:
I get that a superclass might not fit well, but I think a helper class to encapsulate the resend code could help clean up the code, and perhaps there other cases where it would be useful.

Reply:
I think in that case its not worth it as there are small differences and would not make the code better readable if important parts are delegated to another class. If you like you can try it out and send me a patch, but I did a file compare and did not had the impression avoiding some code duplication justifies the higher complexity of distributed code. A part is already in the base class...

Reply:
Then I'll drop it. Main reason was to encapsulate the logic to not have to think about it twice. If we do one more resend message it will likely be worth it.